velox/dwio/parquet/reader/PageReader.cpp

```diff
@@ -364,6 +365,32 @@ void PageReader::prepareDictionary(const PageHeader& pageHeader) {
           // We start from the end to allow in-place expansion.
           values[i] = parquetValues[i];
         }
+      } else if (
+          parquetType == thrift::Type::INT64 &&
+          type_->logicalType_.has_value()) {
+        auto logicalType = type_->logicalType_.value();
+        if (logicalType.__isset.TIMESTAMP) {
+          VELOX_CHECK(
+              logicalType.TIMESTAMP.isAdjustedToUTC,
+              "Only UTC adjusted Timestamp is supported");
+          auto values = dictionary_.values->asMutable<Timestamp>();
+          auto parquetValues = dictionary_.values->asMutable<char>();
+          int64_t units;
+
+          if (logicalType.TIMESTAMP.unit.__isset.MICROS) {
+            for (auto i = dictionary_.numValues - 1; i >= 0; --i) {
+              memcpy(&units, parquetValues + i * typeSize, typeSize);
+              values[i] = util::fromUTCMicros(units);
+            }
+          } else if (logicalType.TIMESTAMP.unit.__isset.MILLIS) {
+            for (auto i = dictionary_.numValues - 1; i >= 0; --i) {
+              memcpy(&units, parquetValues + i * typeSize, typeSize);
+              values[i] = util::fromUTCMillis(units);
+            }
+          } else {
+            VELOX_NYI("Unsupported timestamp unit");
```

Comment:
nit: It seems only nano is unsupported. Maybe something like `Nano timestamp unit is not supported.` could be clearer.